### PR TITLE
test(dataset): unit test for listDataSetsForProject

### PR DESCRIPTION
**Pull request by janeyuaws**
Issue #, if available: CHAM-221

Description of changes:

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Tests covered:

* Role contains ITAdmin
    * role is only ITAdmin
    * role includes ITAdmin and ProjectAdmin
* paginationToken gets returned if mock of workbenchDataSetService returns a pagination token
* filtering by project works
     * when results from workbenchDataSetService is greater than pageSize
     * when results from workbenchDataSetService is less or equal to pageSize
* custom generation of pagination token

### CODE REVIEW

File: solutions/swb-reference/src/services/dataSetService.test.ts [hunk 1]
```diff
@@ -684,4 +684,207 @@ describe('DataSetService', () => {
...
+            mockUser,
+            pageSize,
+            paginationToken
+          );
+          // CHECK
+          expect(actualResponse).toEqual(mockResponse);
+        });
+      });
+
+      describe('role contains ITAdmin and ProjectAdmin', () => {
```
Comment (aws-tyler) [inline solutions/swb-reference/src/services/dataSetService.test.ts:724]: :tada: great idea for a test. i don't think we've got many multi-role ones.

**Review by aws-tyler — CHANGES_REQUESTED** [review]
Looks good! I just want to double check with you about the tests around the pagination token before I approve.

File: solutions/swb-reference/src/services/dataSetService.test.ts [hunk 2]
```diff
@@ -684,4 +684,207 @@ describe('DataSetService', () => {
...
+            pageSize,
+            randomInputToken
+          );
+          // CHECK
+          expect(actualResponse.paginationToken).toEqual(mockReturnedToken);
+        });
+      });
+
+      // check custom input token works
+      describe('when custom generation of pagination token', () => {
```
[the remark this answers is not in the document]
Comment (janeyuaws) [reply solutions/swb-reference/src/services/dataSetService.test.ts:863]: Yeah. Need to confirm with either you or @kpark277 what does custom pagination token means.

File: solutions/swb-reference/src/services/dataSetService.test.ts (hunk 1 above)
Comment (janeyuaws) [reply solutions/swb-reference/src/services/dataSetService.test.ts:724]: Doesn’t hurt to test. The role is a list indicating we could have multiple roles, right?

Comment (janeyuaws) [reply solutions/swb-reference/src/services/dataSetService.test.ts:724]: Someone be a pa for one proj and researcher for another

File: solutions/swb-reference/src/services/dataSetService.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (janeyuaws) [reply solutions/swb-reference/src/services/dataSetService.test.ts]: updated after confirming what's 'custom pagination': checking if we don't go through the entire result set from ddb, we create our own pagination token and return that

File: solutions/swb-reference/src/services/dataSetService.test.ts [hunk 3]
```diff
@@ -859,29 +859,48 @@ describe('DataSetService', () => {
...
-              data: [mockDataSet], // have one result to avoid infinite loop
-              paginationToken: customToken
-            };
+          mockDataSetListWithID = [
+            DataSetParser.parse({ ...mockDataSetWithoutID, id: 'dataSetId1' }),
+            DataSetParser.parse({ ...mockDataSetWithoutID, id: 'dataSetId2' })
+          ];
+
+          pageSize = 1;
+          console.log(mockDataSetListWithID);
```
Comment (aws-tyler) [inline solutions/swb-reference/src/services/dataSetService.test.ts:883]: still need this?

**Review by aws-tyler — APPROVED** [review]

**Pull request merged**